Storybook v5

BPScott opened the pull request:
### WHY are these changes introduced?

Dat bleeding edge

### WHAT is this pull request doing?

Updating storybook to version 5

### How to 🎩

- Try it out locally and at https://polaris-react-pr-1140.herokuapp.com
- Check shrink-ray and percy still work as you'd expect
- Check to see if the preview still looks good on a mobile device

Also the Accessibility audits work now!

kaelig commented:
This is great!!!

Found a few bugs:

---

This button links to https://polaris-react-pr-1140.herokuapp.com/iframe, which shows "no preview".

![Screenshot 2019-04-03 16 25 15](https://user-images.githubusercontent.com/85783/55519540-18236280-562d-11e9-9af4-ee93c94804b5.png)

---

When resizing the browser from small to large or large to small viewport, this error message appears:

```
Warning: Failed prop type: Invalid prop `active` of type `boolean` supplied to `<<anonymous>>`, expected `number`.
    in Unknown (created by MobileLayout)
    in MobileLayout (created by t)
    in t
    in Unknown
    in Unknown (created by Manager)
    in ThemeProvider (created by Manager)
    in Manager (created by Context.Consumer)
    in L (created by QueryLocation)
    in QueryLocation (created by Root)
    in t (created by Root)
    in HelmetProvider (created by Root)
    in Root
```
(not sure it's a big issue)

---

Clicking the "toggle background grid" button doesn't seem to be doing anything.

![Screenshot 2019-04-03 16 27 14](https://user-images.githubusercontent.com/85783/55519596-5d479480-562d-11e9-9ce2-beb841a7926d.png)

BPScott commented:
> This button links to https://polaris-react-pr-1140.herokuapp.com/iframe, which shows "no preview".

Serve was trying to be helpful and rewrite *.html urls. I've told it to cut that out.

> Clicking the "toggle background grid" button doesn't seem to be doing anything.

Polaris sets a background color on the `<body>` of the preview area which sat on top of the grid so you couldn't see it. I've fixed that up now by overriding that behaviour and adding the backgrounds addon which lets us toggle backgrounds.

> When resizing the browser from small to large or large to small viewport, this error message appears [snip]

Looks like a problem with storybook https://github.com/storybooks/storybook/pull/6241. Doesn't seem to affect anything so happy to ignore that for now

Can you check those two issues again and see if they're fixed (the redirect one might require you kicking your network cache if those redirects were cached)

I've also added the responsive addon so you can resize the preview area.
